plugins/catalog-import/src/util/urls.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import parseGitUri from 'git-url-parse';
+
+export type UrlType = 'file' | 'repo';
```

Comment:
`'file'` should be `'blob'` right?

What does `'repo'` actually represent? 😅

Reply:
We can just choose some convention - sure, `blob` is what the underlying lib uses.

The semantics of the other one is unclear still. The plugin is under development and this is subject to change. The original code looked at the url type to see if the user had pasted exactly a repo url (no filepath at all under it), and then triggering the import code flow. But the code actually should support both a repo root and a folder within a repo.

So essentially ... best effort, to be iterated on.

Reply:
Still a bit confused though, is `'repo'` actually a possible output of `parseGitUri`?

Reply:
Supposedly, yes.

Reply:
`The type of filepath in the url ("blob" or "tree").`

https://github.com/IonicaBizau/git-url-parse

But it's all so sloppy in this lib - it's sometimes filled in, sometimes not, sometimes things are suddenly the empty string. So the code I added was looking for only the exact match of `'repo'` and nothing else.